### DISCUSSION TITLE
Suboptimal routing creates an exit from a freeway to cut across land+roadways to reach a POI

### DISCUSSION THREAD

**Discussion by mirror176** [osmandapp/OsmAnd#13608] [Maps Related]
Was routing to https://www.openstreetmap.org/way/347132479 from the southeast. Osmand from fdroid (says it was updated 3 weeks ago) and current Arizona map both normally and after live update shows the same issue. Osmand approaches it from the east with a final link going from Aqua Fria Freeway and cutting across the onramp to Aqua Fria Freeway and the serviceway behind the building to indicate reaching the destination.
  When driving we didn't take the freeway and went through West Union Hills Drive to the north to reach North 83rd Avenue to the west and then entered the parking lot just west of the POI to avoid the potentially slow parking lot drive if we entered it from the north. Currently shown with https://www.openstreetmap.org/directions?engine=graphhopper_car&route=33.64825%2C-112.22728%3B33.64904%2C-112.23461 which isn't starting location I was at but represents this part of the routing. We also approached the area along North 83rd Avenue from the south with routing saying to take the freeway+jump over to the POI in the same way.
  Clicking on the final leap and choosing 'avoid road' results in a new calculation and doing the same jump with the avoid road symbol above it. Avoiding the Aqua Fria Freeway segment that it jumps from permits the more desired form.
  As someone new to Osmand, is there a way to apply other routers like OSRM and GraphHopper to Osmand and are there easy ways to review/edit steps of the route with both text+map of the route shown at the same time?
  I didn't find them with searching but should the route 'hopping' and avoiding the final hop be submitted as new bugs? Presume the rest of my undesired view of the route may be related to traffic lights downgrading routes too far as I read in other posts.

**Comment by vshcherb** [comment]
OSRM https://www.openstreetmap.org/directions?engine=fossgis_osrm_car&route=33.64825%2C-112.22728%3B33.64904%2C-112.23461#map=17/33.65055/-112.23008 produces exactly the same route as OsmAnd, so I'm not sure what's the issue is
